autonat.go

```diff
@@ -61,12 +63,18 @@ type AmbientAutoNAT struct {
 }
 
 // NewAutoNAT creates a new ambient NAT autodiscovery instance attached to a host
-func NewAutoNAT(ctx context.Context, h host.Host) AutoNAT {
+func NewAutoNAT(ctx context.Context, h host.Host, ga ...GetAddrs) AutoNAT {
+	getAddrs := h.Addrs
+	if len(ga) > 0 {
```

Comment:
Is there a reason we decided against the customary `Option` approach? I prefer that approach, as this appears very limiting in the future.

Reply:
turned into static argument per Magik's suggestion.